### PR TITLE
Develop to next post 1.5.11

### PR DESCRIPTION
**Pull request by glennmatthews**
# Closes: #n/a
# What's Changed

Periodic code sync from `develop` up into `next`. 

# TODO

- [x] Explanation of Change(s)
- n/a Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- n/a Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- n/a Example Plugin Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/circuits/tests/test_filters.py [hunk 1]
```diff
@@ -37,7 +37,6 @@ def setUpTestData(cls):
 
         providers = Provider.objects.all()[:2]
         circuit_types = CircuitType.objects.all()[:2]
-        cls.regions = Region.objects.filter(sites__isnull=False, children__isnull=True, parent__isnull=True)[:2]
```
Comment (glennmatthews) [inline nautobot/circuits/tests/test_filters.py:40]: The changes to this file aren't from `develop`, they're just some leftovers I noticed while troubleshooting merge conflicts.

File: nautobot/core/models/name_color_content_types.py [hunk 2]
```diff
@@ -97,6 +97,6 @@ def to_csv(self):
             self.name,
             self.slug,
             self.color,
-            f'"{self.get_content_types()}"',  # Wrap labels in double quotes for CSV
+            self.get_content_types(),
```
Comment (glennmatthews) [inline nautobot/core/models/name_color_content_types.py:100]: This was wrong code uncovered by the new `test_queryset_to_csv` test :+1:

File: nautobot/extras/tests/test_filters.py [hunk 3]
```diff
@@ -248,44 +257,66 @@ def test_role(self):
...
     def test_type(self):
-        device_types = self.device_types[:2]
-        params = {"device_type_id": [device_types[0].pk, device_types[1].pk]}
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), len(device_types))
-        params = {"device_type": [device_types[0].slug, device_types[1].slug]}
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), len(device_types))
+        device_types = list(self.device_types[:2])
+        filter_params = [
+            {"device_type_id": [device_types[0].pk, device_types[1].pk]},
+            {"device_type": [device_types[0].pk, device_types[1].slug]},
```
Comment (glennmatthews) [inline nautobot/extras/tests/test_filters.py:263]: These should certainly be changed over in `next` to use `generic_filter_tests` at some point but that's beyond the scope of this PR.

File: nautobot/ipam/tests/test_filters.py [hunk 4]
```diff
@@ -81,33 +81,29 @@ def test_enforce_unique(self):
...
-        )
-        params = {"import_target": [route_targets[0].name, route_targets[1].name]}
-        self.assertQuerysetEqual(
-            self.filterset(params, self.queryset).qs,
-            self.queryset.filter(import_targets__in=route_targets).distinct(),
-            ordered=False,
-        )
+        filter_params = [
+            {"import_target_id": [route_targets[0].pk, route_targets[1].pk]},
+            {"import_target": [route_targets[0].pk, route_targets[1].name]},
```
Comment (glennmatthews) [inline nautobot/ipam/tests/test_filters.py:86]: More future `generic_filter_tests` candidates.

File: nautobot/virtualization/tests/test_filters.py [hunk 5]
```diff
@@ -218,18 +218,28 @@ def test_location(self):
...
     def test_cluster_group(self):
-        cluster_groups = ClusterGroup.objects.all()[:2]
-        params = {"cluster_group_id": [cluster_groups[0].pk, cluster_groups[1].pk]}
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 2)
-        params = {"cluster_group": [cluster_groups[0].slug, cluster_groups[1].slug]}
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 2)
+        cluster_groups = list(ClusterGroup.objects.all()[:2])
+        filter_params = [
+            {"cluster_group_id": [cluster_groups[0].pk, cluster_groups[1].pk]},
+            {"cluster_group": [cluster_groups[0].pk, cluster_groups[1].slug]},
```
Comment (glennmatthews) [inline nautobot/virtualization/tests/test_filters.py:224]: More `generic_filter_tests` candidates.

**Review by bryanculver — APPROVED** [review]

**Pull request merged**